ludwig/backend/ray.py

```diff
@@ -366,13 +395,45 @@ def spread_env(use_gpu: bool = False, num_workers: int = 1, **kwargs):
 def create_runner(**kwargs):
     trainer_kwargs = get_trainer_kwargs(**kwargs)
     with spread_env(**trainer_kwargs):
-        trainer = Trainer(**trainer_kwargs)
+        yield RayAirRunner(trainer_kwargs)
 
-    trainer.start()
-    try:
-        yield trainer
-    finally:
-        trainer.shutdown()
+
+class RayAirRunner:
+    def __init__(self, trainer_kwargs: Dict[str, Any]) -> None:
+        trainer_kwargs = copy.copy(trainer_kwargs)
+        trainer_kwargs.pop("backend", None)
+
+        # When training on GPU, you want to pack workers together to limit network latency during
+        # allreduce. Conversely, for CPU training you want to spread out the workers to limit
+        # CPU and memory contention.
+        strategy = "PACK" if trainer_kwargs.get("use_gpu") else "SPREAD"
+        self.scaling_config = ScalingConfig(placement_strategy=strategy, **trainer_kwargs)
+
+    def run(
+        self,
+        train_loop_per_worker: Callable,
+        config: Dict[str, Any],
+        dataset: Dict[str, Any],
+        callbacks: List[Any] = [],
+    ) -> List[Any]:
+        trainer = HorovodTrainer(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config=config,
+            datasets=dataset,
+            scaling_config=self.scaling_config,
+            dataset_config={
+                "train": DatasetConfig(split=True),
+                "val": DatasetConfig(split=True),
+                "test": DatasetConfig(split=True),
+            },
+            # run_config=RunConfig(
+            #     callbacks=callbacks,
+            #     verbose=2,
+            # ),
+            # resume_from_checkpoint=None,
+        )
+        # tuner = ray.tune.Tuner(trainer)
```

Comment:
Only way to pass in callbacks is through the RunConfig, but that internally starts saying that the trainer needs to be wrapped in the `ray.tune.Tuner()` call, likely because callbacks are shared between the trainer and ray tune in Ray 2.0. Once I do this though, I keep seeing this error: 

```
ray.tune.error.TuneError: Tune run failed. Please use tuner = Tuner.restore("/home/vscode/ray_results/HorovodTrainer_2022-10-26_16-33-44") to resume.
```